tmt/result.py

```diff
@@ -302,31 +322,54 @@ def interpret_result(self, interpret: ResultInterpret) -> 'Result':
         :returns: :py:class:`Result` instance containing the updated result.
         """
 
-        if interpret in (ResultInterpret.RESPECT, ResultInterpret.CUSTOM):
+        if interpret == ResultInterpret.CUSTOM:
+            return self
+
+        # Check for failed checks
+        checks_failed = any(check.result == ResultOutcome.FAIL for check in self.check)
+
+        if interpret == ResultInterpret.RESPECT:
+            if self.respect_checks and checks_failed:
+                self.result = ResultOutcome.FAIL
+                if self.note:
+                    self.note += ', check failed'
+                else:
+                    self.note = 'check failed'
+            return self
+
+        # Handle XFAIL
+        if interpret == ResultInterpret.XFAIL:
+            if self.result == ResultOutcome.PASS and checks_failed:
+                self.result = ResultOutcome.FAIL
+                if self.note:
+                    self.note += ', check failed'
+                else:
+                    self.note = 'check failed'
+            elif self.result == ResultOutcome.FAIL:
+                self.result = ResultOutcome.PASS
+                if self.note:
+                    self.note += ', expected failure'
+                else:
+                    self.note = 'expected failure'
+            elif self.result == ResultOutcome.PASS:
+                self.result = ResultOutcome.FAIL
+                if self.note:
+                    self.note += ', unexpected pass'
+                else:
+                    self.note = 'unexpected pass'
             return self
 
         # Extend existing note or set a new one
         if self.note:
             self.note += f', original result: {self.result.value}'
-
```

Comment:
Please, keep the whitespace for those of us who don’t enjoy dense code :)

Reply:
Don't remember doing this. autopep8?  
In any case, I'll clean things up before being ready for review.